src/Framework/Receipts/Actions/GenerateConfirmationPageReceipt.php

```diff
@@ -249,6 +249,16 @@ private function fillSettings(DonationReceipt $receipt)
 
         $receipt->settings->addSetting('currency', $receipt->donation->amount->getCurrency()->getCode());
         $receipt->settings->addSetting('donorDashboardUrl', get_permalink(give_get_option('donor_dashboard_page')));
+
+        if (function_exists('give_pdf_receipts')) {
+            $link = sprintf(
+                '<a id="give-pdf-receipt-link" title="%3$s" href="%1$s">%2$s</a>',
+                give_pdf_receipts()->engine->get_pdf_receipt_url($receipt->donation->id),
+                give_pdf_receipts_download_pdf_text(false),
+                give_pdf_receipts_download_pdf_text(false)
+            );
+            $receipt->settings->addSetting('pdfReceiptLink', $link);
+        }
```

Comment:
@glaubersilva in general I try to avoid using add-on code directly in givewp core as it can be unpredictable when it changes. i'm thinking instead we can do something like this:

```
$receipt->settings->addSetting(
    'pdfReceiptLink',
    apply_filters('givewp_confirmation_page_receipt_settings_pdfReceiptLink', '', $receipt)
);
```

Then in the PDF receipts addon:

```
add_filter('givewp_confirmation_page_receipt_settings_pdfReceiptLink', function ($receipt) {
    return sprintf(
    '<a id="give-pdf-receipt-link" title="%3$s" href="%1$s">%2$s</a>',
        give_pdf_receipts()->engine->get_pdf_receipt_url($receipt->donation->id),
        give_pdf_receipts_download_pdf_text(false),
        give_pdf_receipts_download_pdf_text(false)
    );
});
```

This way, the setting is empty until the add-on is installed.

Eventually, we'll add additional actions so add-ons can extend the receipt - but I think filters are better for settings as they are so specific as a known set of key/values.

Reply:
Resolved in https://github.com/impress-org/give-pdf-receipts/pull/249